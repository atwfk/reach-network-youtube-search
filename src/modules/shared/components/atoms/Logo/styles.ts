import styled from "styled-components";
import { SCREENS } from "../../../constants";

export const StyledLogo = styled.div`
  position: relative;
  margin-right: 10px;

  @media (min-width: ${SCREENS.MD}) {
    margin-right: 20px;
  }

  span {
    position: absolute;
    display: none;
    top: 0;
    right: 20px;
    color: #9d9d9d;
    font-size: 12px;
    @media (min-width: ${SCREENS.MD}) {
      display: block;
    }
  }

  .primary-logo {
    width: 180px;
    height: 40px;
    display: none;
    @media (min-width: ${SCREENS.MD}) {
      display: block;
    }
  }

  .secondary-logo {
    width: 60px;
    height: 40px;
    background-color: red;
    display: block;
    @media (min-width: ${SCREENS.MD}) {
      display: none;
    }
  }
`;
