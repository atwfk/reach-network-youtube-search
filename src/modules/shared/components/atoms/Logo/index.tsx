import React from "react";
import type { FC, ReactElement } from "react";
import { ReactComponent as PrimaryLogo } from "../../icons/primary-logo.svg";
import { ReactComponent as SecondaryLogo } from "../../icons/secondary-logo.svg";
import { StyledLogo } from "./styles";

const Logo: FC = (): ReactElement => {
  return (
    <StyledLogo>
      <span>EG</span>
      <PrimaryLogo className="primary-logo" />
      <SecondaryLogo className="secondary-logo" />
    </StyledLogo>
  );
};

export default Logo;
