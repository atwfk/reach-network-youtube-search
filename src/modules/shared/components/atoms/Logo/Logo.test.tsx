import renderer from "react-test-renderer";
import Logo from ".";

describe("Logo Component", () => {
  it("Should Logo with primary and secondary", () => {
    const tree = renderer.create(<Logo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
