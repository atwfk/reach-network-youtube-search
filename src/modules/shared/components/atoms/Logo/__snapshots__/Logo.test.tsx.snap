// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component Should Logo with primary and secondary 1`] = `
<div
  className="sc-bdvvtL ibexgz"
>
  <span>
    EG
  </span>
  <svg
    className="primary-logo"
  >
    primary-logo.svg
  </svg>
  <svg
    className="secondary-logo"
  >
    secondary-logo.svg
  </svg>
</div>
`;
